lib/dotcom_web/components/trip_planner/trip_planner_results_section.ex

```diff
@@ -39,19 +41,43 @@ defmodule DotcomWeb.Components.TripPlanner.TripPlannerResultsSection do
           </button>
         </div>
       </.async_result>
-
-      <.live_component
-        module={MbtaMetro.Live.Map}
-        id="trip-planner-map"
-        class={[
-          "h-64 md:h-96 w-full",
-          "relative overflow-none row-span-2",
-          @itinerary_selection == :summary && "hidden md:block"
-        ]}
-        config={@map_config}
-        pins={[@from, @to]}
-      />
-
+      <.async_result :let={results} assign={@results}>
+        <%= if results && @itinerary_selection != :summary do %>
+          <% {:detail,
+           %{itinerary_group_index: itinerary_group_index, itinerary_index: itinerary_index}} =
+            @itinerary_selection %>
+          <% itinerary =
```

Comment:
You can move a bunch of this logic out of the HEEx template itself and into the body of `&trip_planner_results_section/1` right before the HEEx template by doing something like:

```elixir
def trip_planner_results_section(
      %{itinerary_selection: itinerary_selection, results: itinerary_results} = assigns
    ) do
  assigns =
    if itinerary_results.result && itinerary_selection != :summary do
      {:detail,
       %{itinerary_group_index: itinerary_group_index, itinerary_index: itinerary_index}} =
        itinerary_selection

      itinerary =
        Enum.at(itinerary_results.result, itinerary_group_index).itineraries
        |> Enum.at(itinerary_index)

      itinerary_map = TripPlan.Map.itinerary_map(itinerary)
      lines = TripPlan.Map.get_lines(itinerary_map)
      points = TripPlan.Map.get_points(itinerary_map)

      assign(assigns, %{lines: lines, points: points})
    else
      assign(assigns, %{lines: [], points: []})
    end

  ~H"""
  <!-- ... -->
  """
end
```

(I'm not attached to ☝️ code exactly as-is - I think it could be cleaned up a bit, although I would buy the argument that polishing this chunk of code isn't worth the effort)

That lets you have `lines` and `points` be `[]` unless there's an itinerary powering real data for them, which then lets you just unconditionally render the map, with or without the itinerary data:

```heex
<.live_component
  module={MbtaMetro.Live.Map}
  id="trip-planner-map"
  class={[
    "h-64 md:h-96 w-full",
    "relative overflow-none row-span-2",
    @itinerary_selection == :summary && "hidden md:block"
  ]}
  config={@map_config}
  lines={@lines}
  pins={[@from, @to]}
  points={@points}
/>
```

...and then get rid of _all_ of the extra logic that's embedded inside the HEEx blob.

That also has the advantage of fixing [this bug](https://github.com/mbta/dotcom/pull/2258#issuecomment-2532661926), by bringing the map back outside of the `<.async_result />`.